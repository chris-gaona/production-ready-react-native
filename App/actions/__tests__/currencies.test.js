import { getInitialConversion, changeCurrencyAmount } from '../currencies'

describe('getInitialConversion', () => {
  it('should create a properly formatted action', () => {
    const expected = { type: 'GET_INITIAL_CONVERSION' }
    const actual = getInitialConversion()
    expect(actual).toEqual(expected)
  })
})

describe('changeCurrencyAmount', () => {
  it('should create an action with an amount of 100', () => {
    const expected = { type: 'CHANGE_CURRENCY_AMOUNT', amount: 100 }
    const actual = changeCurrencyAmount(100)
    expect(actual).toEqual(expected)
  })

  it('should convert a string value to a float', () => {
    const expected = { type: 'CHANGE_CURRENCY_AMOUNT', amount: 100 }
    const actual = changeCurrencyAmount('100')
    expect(actual).toEqual(expected)
  })
})
