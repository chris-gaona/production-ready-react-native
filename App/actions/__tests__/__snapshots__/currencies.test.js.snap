// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changeCurrencyAmount should convert a string value to a float 1`] = `
Object {
  "amount": 100,
  "type": "CHANGE_CURRENCY_AMOUNT",
}
`;

exports[`changeCurrencyAmount should create an action with an amount of 100 1`] = `
Object {
  "amount": 100,
  "type": "CHANGE_CURRENCY_AMOUNT",
}
`;

exports[`getInitialConversion should create a properly formatted action 1`] = `
Object {
  "type": "GET_INITIAL_CONVERSION",
}
`;
